Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/modules/registeredtest/TestResultDonor.kt

```diff
@@ -0,0 +1,214 @@
+package de.rki.coronawarnapp.datadonation.analytics.modules.registeredtest
+
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.datadonation.analytics.modules.DonorModule
+import de.rki.coronawarnapp.datadonation.analytics.storage.AnalyticsSettings
+import de.rki.coronawarnapp.risk.RiskLevelSettings
+import de.rki.coronawarnapp.risk.storage.RiskLevelStorage
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.formatter.TestResult
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import org.joda.time.Instant
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TestResultDonor @Inject constructor(
+    private val analyticsSettings: AnalyticsSettings,
+    private val appConfigProvider: AppConfigProvider,
+    private val riskLevelSettings: RiskLevelSettings,
+    private val riskLevelStorage: RiskLevelStorage,
+    private val timeStamper: TimeStamper,
+) : DonorModule {
+
+    override suspend fun beginDonation(request: DonorModule.Request): DonorModule.Contribution {
+        val scannedAfterConsent = analyticsSettings.testScannedAfterConsent.value
+        if (!scannedAfterConsent) {
+            Timber.d("Skipping TestResultMetadata donation (testScannedAfterConsent=%s)", scannedAfterConsent)
+            return TestResultMetadataNoContribution
+        }
+
+        val timestampAtRegistration = LocalData.initialTestResultReceivedTimestamp()
```

Comment:
As we want to get rid of local data, we should avoid further usage and store the needed data in a dedicated and separate space.

Reply:
I used this based on a discussion with Max, refactoring is a different topic. When is refactored it will replaced here too. Additionally I don't see any reason to create a new one saving the same value

Reply:
You could just add it to the other data you collect. It makes it easy to wipe data without side effects.

Reply:
It is not related to analytics data, I don't really like we should create a new one for every feature that uses it